ghedesigner/ghe/rowwise.py

```diff
@@ -1,9 +1,201 @@
+from functools import lru_cache
 from math import atan, cos, inf, pi, sin, sqrt
 
 import numpy as np
 
 from ghedesigner.constants import DEG_TO_RAD, PI_OVER_2, RAD_TO_DEG
-from ghedesigner.ghe.shape import Shapes, sort_intersections
+from ghedesigner.ghe.shape import Shapes, point_polygon_check, sort_intersections
+
+NEIGHBORHOOD_OFFSETS = ((-1, -1), (-1, 0), (-1, 1),
+                        (0, -1), (0, 0), (0, 1),
+                        (1, -1), (1, 0), (1, 1))
+
+
+class DeferredDuplicateCheckList:
+
+    def __init__(self, spacing):
+
+        self.points = {}
+        self.buckets = {}
+        self.points_to_check = []
+        self.spacing = spacing
+        self.partitioned = False
+        self.proximity_checks_found = False
+        self.largest_index = 0
+        self.bucket_keys = None
+
+    def _get_bucket_key(self, point):
+        return int(point[0] // self.spacing), int(point[1] // self.spacing)
+
+    def _get_bucket_key(self, px, py):
+        return int(px // self.spacing), int(py // self.spacing)
+
+    def _get_bucket_keys(self, points_array):
+        return (points_array / self.spacing).astype(int)
```

Comment:
@TNWest thanks for working on this. Python doesn't have a clean way to support overloaded functions, as you have here. Can we trim this down to a single function or use different function names?